src/fft.jl

```diff
@@ -0,0 +1,576 @@
+using AbstractFFTs
+using LinearAlgebra
+using Dagger: DArray, @spawn, InOut, In
+using MPI
+
+struct FFT end
+struct RFFT end
+struct IRFFT end
+struct IFFT end
+struct FFT! end
+struct RFFT! end
+struct IRFFT! end
+struct IFFT! end
+
+export FFT, RFFT, IRFFT, IFFT, FFT!, RFFT!, IRFFT!, IFFT!, fft, ifft
```

Comment:
```suggestion
```

We shouldn't need to export any of these. If users want to use Dagger's FFT, all they should need to do is `using Dagger, AbstractFFTs` and then `fft(A)` where `A <: DArray`.